src/main/java/supercoder79/chunkpregen/PregenerationTask.java

```diff
@@ -86,7 +86,16 @@ private void tryEnqueueTasks() {
                 return;
             }
 
-            int enqueueCount = BATCH_SIZE - this.queuedCount.get();
+            int queuedCount = this.queuedCount.get();
```

Comment:
I think this might cut off chunk generation at the wrong point - the total chunk count is calculated based on the given radius, but the iterator gives a bit of a bigger radius due to the coarse cells. Might make sense to clip the coarse iterator results to the actual radius? 🙂 Alternatively, we could pull the count from the iterator

Reply:
Strangely in my 32 radius test, it didn't seem to cut off. I'll have to do more testing. But yeah I'll poke at the iterator and see what I can do there

![image](https://github.com/jaskarth/fabric-chunkpregenerator/assets/40846040/44e14d2a-2e35-4d1f-ad30-c6ca20791da3)

Reply:
Thank you! Yeah the check on the result of the iterator seemed to have done the trick and doesn't seem to chop off in my testing. Much cleaner solution